components/checkbox/index.css

```
@@ -24,9 +157,299 @@ governing permissions and limitations under the License.
+      &:before {
+        border-color: var(--highcontrast-checkbox-color-default, var(--mod-checkbox-invalid-color-default, var(--spectrum-checkbox-invalid-color-default)));
+      }
+    }
+
+    &.is-indeterminate .spectrum-Checkbox-input:focus-ring,
+    .spectrum-Checkbox-input:focus-ring {
+      & + .spectrum-Checkbox-box {
+        &:before {
+          border-color: var(--highcontrast-checkbox-color-hover, var(--mod-checkbox-invalid-color-hover, var(--spectrum-checkbox-invalid-color-hover)));
+        }
+      }
+    }
+
+    &:hover {
+      .spectrum-Checkbox-input:checked + .spectrum-Checkbox-box,
+      .spectrum-Checkbox-box {
+        &:before {
+          border-color: var(--highcontrast-checkbox-color-hover, var(--mod-checkbox-invalid-color-hover, var(--spectrum-checkbox-invalid-color-hover)));
+        }
+      }
+    }
+  }
+
+  /*
+   * Read-Only
+   *
+   * readonly is not a valid attribute for input[type="checkbox"]
+   * so we borrow the immutability of a disabled checkbox
+   * while using the colors of a default checkbox
+   */
+  &.is-readOnly {
+    border-color: var(--highcontrast-checkbox-color-default, var(--mod-checkbox-control-selected-color-default, var(--spectrum-checkbox-control-selected-color-default)));
+
+    &:hover {
+      .spectrum-Checkbox-box {
+        &:before {
+          border-color: var(--highcontrast-checkbox-color-default, var(--mod-checkbox-control-selected-color-default, var(--spectrum-checkbox-control-selected-color-default)));
+        }
+      }
+    }
+
+    &:active {
+      .spectrum-Checkbox-box {
+        &:before {
+          border-color: var(--highcontrast-checkbox-selected-color-default, var(--mod-checkbox-control-selected-color-default, var(--spectrum-checkbox-control-selected-color-default)));
+        }
+      }
+    }
+  }
+
+  &.is-readOnly .spectrum-Checkbox-input,
+  &.is-readOnly .spectrum-Checkbox-input:checked {
+    &:disabled + .spectrum-Checkbox-box {
+      &:before {
+        border-color: var(--highcontrast-checkbox-color-default, var(--mod-checkbox-control-selected-color-default, var(--spectrum-checkbox-control-selected-color-default)));
+        background-color: var(--highcontrast-checkbox-background-color-default, var(--mod-checkbox-checkmark-color, var(--spectrum-checkbox-checkmark-color)));
+      }
+    }
+
+    &:disabled ~ .spectrum-Checkbox-label {
+      forced-color-adjust: none;
+      color: var(--highcontrast-checkbox-color-default, var(--mod-checkbox-content-color-default, var(--spectrum-checkbox-content-color-default)));
+    }
+  }
+
+  /* Indeterminate */
+  &.is-indeterminate {
+    .spectrum-Checkbox-box,
+    .spectrum-Checkbox-input:checked + .spectrum-Checkbox-box {
+      &:before {
+        border-color: var(--highcontrast-checkbox-highlight-color-default, var(--mod-checkbox-control-selected-color-default, var(--spectrum-checkbox-control-selected-color-default)));
+        border-width: calc(var(--spectrum-checkbox-control-size) / 2);
```

Two things:
1. Can we abstract this calculation out to its own custom property? Perhaps something like `--spectrum-checkbox-checked-border-width` would work (or if you have a better name)?
2. Once you abstract it out, can you also add the `--mod-` custom property here, too?
```suggestion
        border-width: var(--mod-checkbox-checked-border width, var(--spectrum-checkbox-checked-border-width));
```